### PR TITLE
Minor optimisations 3

### PR DESCRIPTION
**Pull request by NotStirred**
Some minor optimisations to the map view loading.
Also fixed not using `SimpleChunkData` when possible

Draft because I'm going to do some more profiling before pushing another change

### CODE REVIEW

**Comment by NotStirred** [comment]
I've been testing the performance of `ChunkData#clear`, as it's called once per-chunk for both the map screen and loading into the octree. (The chunkdata object is reused instead of reallocated)

On 3 JREs I profiled 5 different clearing approaches.
The fastest on all JREs was `System.arraycopy` from a static array of the same size as the blocks/biomes array initialised with all zeros. (The static arrays use about `0.25MiB` together)
Fast copy is an implementation where the first element of the array is set to `0` and is then copied using `System.arraycopy` in a binary fashion throughout the array until full

Zulu8
```
1. System.arraycopy (from static array) average: 0.2749
2. Fast copy average: 0.2835
3. For loop copy average: 0.2865
4. Arrays.Fill average: 0.2892
5. master average: 1.5826
```
openjdk8 hotspot
```
1. System.arraycopy (from static array) average: 0.2636
2. Fast copy average: 0.2766
3. Arrays.Fill average: 0.2786
4. For loop copy average: 0.2789
5. master average: 1.5155
```
openjdk11
```
1. System.arraycopy (from static array) average: 0.2736
2. For loop copy average: 0.2815
3. Fast copy average: 0.2818
4. Arrays.Fill average: 0.2824
5. master average: 0.6643
```

Unless you object to the minor jank @leMaik I'll go with the faster option

**Comment by leMaik** [comment]
@NotStirred Semantically, `Arrays.fill` makes the most sense to me (to fill an array with zeros) and it's almost as fast as the other options.
If you want to go with `System.arraycopy`: Isn't there a special class for empty chunk data already? If so, we could copy the array from there (with a comment on why this is faster, maybe linking your benchmark).

**Comment by NotStirred** [comment]
Sadly `EmptyChunkData` is just that, empty, it returns `0` for all values, and doesn't have any fields

**Comment by leMaik** [comment]
Would be a reasonable place to put the empty array though, if we go with that approach. Not sure if that's worth it (on the other hand… we're talking about 250 KiB here…)

**Comment by NotStirred** [comment]
It's something that will only be used in the `SimpleChunkData` class anyway, which is why I have it there for now
tbh it *really* doesn't matter

**Comment by leMaik** [comment]
As long as it is clear why the code does what it does, it's fine either way. `Arrays.fill` is clear on its own, copying an empty array would need a comment that explains it. I prefer the first for that reason but yeah, we shouldn't overthink it.

**Comment by NotStirred** [comment]
In my testing `SimpleChunkData#clear()` takes up around `1%` of time when loading for the map view, which isn't insignificant

**Comment by leMaik** [comment]
@NotStirred Yes… but the difference between 0.2818 seconds and 0.27… seconds which probably is a microbenchmark prone to errors anyway… it's faster than master, so pick one of them.

**Comment by NotStirred** [comment]
Actually the tests were reasonable, the results I posted were averages over several hundred tests, each consisting of 10k clears of existing `SimpleChunkData` objects. The results should be reliable

**Comment by leMaik** [comment]
Perfect! :+1: As I said, `System.arraycopy` + comment is :100:% fine

File: chunky/src/java/se/llbit/chunky/chunk/SimpleChunkData.java [hunk 1]
```diff
@@ -34,7 +38,7 @@ public SimpleChunkData() {
   }
 
   @Override public int maxY() {
-    return 255;
+    return 256;
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/chunk/SimpleChunkData.java:41]: Why? :smile:

**Review by leMaik — CHANGES_REQUESTED** [review]
Imho the `maxY()` change should be reverted, the rest is great :+1:

**Comment by NotStirred** [comment]
the reason I did the `maxY` change is because with `minY` and `maxY` of `0` inclusive, there is expected to be a block inside the chunkdata, with this change there is not. Basically it works exactly how a list works with a for loop, len `5` actually means `0`-`4`
Though if you want me to revert I can do :)

**Comment by leMaik** [comment]
Oooh… Yes okay, so empty chunks get confusing, I see :sweat_smile: Fine then :+1:

**Comment by NotStirred** [comment]
I had mitigated this previously by having the default min and max at max and min integer, though that was a temporary solution at best and relied on the `ChunkData` implementation to handle it themselves

**Pull request merged**